shell/platform/darwin/ios/ios_surface_factory.h

```diff
@@ -0,0 +1,39 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_DARWIN_IOS_IOS__SURFACE_FACTORY_H_
+#define FLUTTER_SHELL_PLATFORM_DARWIN_IOS_IOS__SURFACE_FACTORY_H_
+
+#include <memory>
+
+#import "flutter/shell/platform/darwin/ios/ios_surface.h"
+#import "flutter/shell/platform/darwin/ios/rendering_api_selection.h"
+
+namespace flutter {
+
+class IOSSurfaceFactory {
+ public:
+  static std::shared_ptr<IOSSurfaceFactory> Create(
+      IOSRenderingAPI rendering_api);
+
+  explicit IOSSurfaceFactory(std::shared_ptr<IOSContext> ios_context);
+
+  ~IOSSurfaceFactory();
+
+  void SetPlatformViewsController(
+      FlutterPlatformViewsController* platform_views_controller);
+
+  std::unique_ptr<IOSSurface> CreateSurface(
```

Comment:
What's the point of a factory that doesn't have virtual methods?  This could just be a function call.

Reply:
Can you elaborate?

Reply:
Yep, right now IOSSurfaceFactory is just an obfuscation on `flutter::IOSSurface::Create`

this:
```
IOSSurfaceFactory factory = IOSSurfaceFactory.create(renderingApi);
factory.setPlatformViewsController(platformViewsController);
return factory.createSurface(layer);
```

could just be this:
```
return CreateSurface(renderingApi, platformViewsController, layer);
```

The factory object doesn't actually add any functionality, it just obscures the logic and worse it creates a mine by storing a raw pointer.  Typically factories are used to allow us to swap out the creation of an object, since this is a c++ class without virtual methods, it can't actually be swapped out.